### ISSUE TITLE
"callbacks" prop should be treated as AMD module

### ISSUE THREAD

**Issue by jhpedemonte** [maqetta/maqetta#2690]
Currently, the "callbacks" property is simply evaluated as JS (through dojo.xhrGet) in _metadata.js_.  Instead, we should treat it as an AMD module, to put it on par with "helper", "tool" and "inlineEdit".

**Comment by peller** [comment]
also, the [dojo,dijit,dojox] callback list should go away and be replaced with actual AMD module references

**Comment by jhpedemonte** [comment]
> also, the [dojo,dijit,dojox] callback list should go away and be replaced with actual AMD module references

What do you mean by this?

**Comment by peller** [comment]
the define() method at the top of the callbacks file lists `dojo, dijit, dojox` as dependencies.  None are proper AMD dependencies.

**Comment by jhpedemonte** [comment]
@peller, I'm confused.  I'm talking about the file `davinci.dojo_1_8/WebContent/metadata/callbacks.js`.  It currently consists of a single closure; it doesn't yet have any AMD syntax.

**Comment by jhpedemonte** [comment]
Oh, I see now.  The source version consists of a function closure.  The **built** version of that _callbacks.js_ file wraps the content with a `define()`:

``` javascript

// wrapped by build app
define("maq-metadata-dojo/callbacks", ["dijit","dojo","dojox"], function(dijit,dojo,dojox){
(function() {
 ...
})();
});
```

Not sure why it does that.  Also, we should add `insertAbsMids: 0` to the Dojo build profiles, so it doesn't generate an MID for the sources.

**Comment by peller** [comment]
oh sorry, yeah, must be some back-compat build magic putting it there then. nevermind.

**Comment by jhpedemonte** [comment]
-> M8

**Comment by jhpedemonte** [comment]
Back to M7.  Fixing this should fix #3117.
